recipes-bsp/u-boot/files/0011-board-siemens-iot2050-get-and-set-m.2-releated-infor.patch

```diff
@@ -0,0 +1,229 @@
+From ba758a41c49507b4abfcf292b695582f8cbb4375 Mon Sep 17 00:00:00 2001
+From: chao zeng <chao.zeng@siemens.com>
+Date: Fri, 7 Jan 2022 16:35:43 +0800
+Subject: [PATCH 11/16] board:siemens:iot2050:get and set m.2 releated
+ information
+
+Get the M.2 configuration,enable the power of the m.2
+Set the releated overlay according to the M.2 configuration
+
+Signed-off-by: chao zeng <chao.zeng@siemens.com>
+---
+ board/siemens/iot2050/board.c | 173 +++++++++++++++++++++++++++++++++-
+ 1 file changed, 171 insertions(+), 2 deletions(-)
+
+diff --git a/board/siemens/iot2050/board.c b/board/siemens/iot2050/board.c
+index 817fbf4ecb..21c82556e9 100644
+--- a/board/siemens/iot2050/board.c
++++ b/board/siemens/iot2050/board.c
+@@ -47,6 +47,141 @@ struct iot2050_info {
+ 
+ DECLARE_GLOBAL_DATA_PTR;
+ 
++#define M2_CONFIG_LEN    4
++#define PCIE_MUX_CONTROL 3
++#define SERDES_SELECT    2
++
++typedef struct gpio_config {
++	char *label;
++	char *gpio_name;
++	uint32_t direction;
++}gpioConfig_t;
++
++typedef enum lane0_config{
++	USB30 = 0,
++	PCIE0_LANE0,
++}lane0_config_e;
++
++typedef enum lane1_config{
++	PCIE1_LANE0 = 0,
++	PCIE0_LANE1,
++}lane1_config_e;
++
++typedef enum key_type{
++	M2_B_KEY = 0,
++	M2_BE_KEY,
++	M2_E_KEY,
++}key_type_e;
++
++typedef struct serdes_config {
++	lane0_config_e  lane0_sel;
++	lane1_config_e  lane1_sel;
++}serdes_config_t;
++
++typedef struct key_config {
++	key_type_e config_desc;    
++	int m2config[M2_CONFIG_LEN];
```

Comment:
better to use bitmap?

Reply:
Same for below `m2_pinmux_value`, one byte is too big to store a bit message.

Reply:
good idea